### DISCUSSION TITLE
Changelog

### DISCUSSION THREAD

**Discussion by lokki11199** [The-OpenROAD-Project/OpenROAD-flow-scripts#1832] [Q&A]
Hello,
I last used ORFS last year and now I want to understand the innovations. I have not found any changelog. Can someone please tell me where to find a list of changes or summarize what has changed.

Thank You!

**Comment by maliberty** [comment]
We don't keep a regular changelog aside from the git history.  Some notable point would be the inclusion of the IHP PDK, addition of post-global-route timing optimization, improved CTS, etc.  Is there any particular area of interest or concern?

**Reply by lokki11199** [reply]
I'm interested in key changes such as you listed above. I am also interested in the following things:
1. Did timing-driven logic synthesis based on sdc appear?
2. Previously ORFS did not support corner-based optimization, is this option available now?
3. Last time I read that LVS is under development and its results may not be correct. What is the status of this now?
4. Is CCS still under development?
5. Has support for SystemVerilog expanded?

Also, is there access to earlier versions of the documentation so I can compare thread variables for a more detailed study? 

Thank you for your answers!

**Comment by maliberty** [comment]
> 1. Did timing-driven logic synthesis based on sdc appear?

Still in development

> 2. Previously ORFS did not support corner-based optimization, is this option available now?

The support in OR has been there for a while but we haven't enabled it in ORFS.  If there is interest we could work on it.

> 3. Last time I read that LVS is under development and its results may not be correct. What is the status of this now?

I don't think LVS was ever in the plan.  KLayout or Magic support it.

> 4. Is CCS still under development?

It is still in development.

> 5. Has support for SystemVerilog expanded?

No change - that is a yosys issue.

**Reply by lokki11199** [reply]
> The support in OR has been there for a while but we haven't enabled it in ORFS. If there is interest we could work on it.

Yes, it would be great if the flow completed with corner-based optimization.

> I don't think LVS was ever in the plan. KLayout or Magic support it.

Then what was meant about LVS here: https://github.com/The-OpenROAD-Project/OpenROAD/discussions/3379?

**Reply by maliberty** [reply]
I don't know what @vijayank88 was referring to unless he is working on it.

Is there a particular PDK you are using?

**Reply by lokki11199** [reply]
I'm trying to add a proprietary PDK, so I'll have to convert LVS rules to Klayout format, but to do that I need to understand how well LVS checking works in ORFS so I don't mistake a bug in the software for a bug in my own LVS rules.

**Reply by rovinski** [reply]
All I can say is good luck, because converting LVS rules is a significant challenge. It would require a lot of checking against a golden model and many tests to be confident that the rule conversion matches 1:1, unless the PDK is a very mature node that doesn't have many rules. I suppose LVS is easier than DRC, but either way, it's difficult.

**Reply by lokki11199** [reply]
I really hope it gets done eventually. Maybe you can give some recommendations on this? 180 nm tech node.

**Reply by maliberty** [reply]
Even if there were some option to do LVS you would still need to setup the KLayout rules yourself for a proprietary PDK.  Almost all of the work is in that step not in enabling it in ORFS.

**Reply by rovinski** [reply]
Being 180nm will make it easier (as opposed to XXnm). If you are starting from a proprietary rule deck, try to do a 1:1 conversion to an equivalent rule in KLayout, if it exists. For LVS, most operations you would need should exist. For DRC, there may not be a 1:1 correspondence in proprietary operations to KLayout operations so you might have to get creative in how to define a check (but I wouldn't foresee much of this in 180nm).

Refer to the KLayout user guide and LVS API guide for which operations are supported and how to use them. Look at examples from other PDKs.

Create lots of examples for validating your rule deck. Manually examine it to verify, or if you have a golden model, compare your results and make sure they match (at least such that full match == full match and non-matching == non-matching; it's very difficult to get two LVS tools to agree exactly how two circuits differ)

**Reply by rovinski** [reply]
> Even if there were some option to do LVS you would still need to setup the KLayout rules yourself for a proprietary PDK. Almost all of the work is in that step not in enabling it in ORFS.

Also true

**Reply by lokki11199** [reply]
Thank you for your replies! I think the discussion can be closed.
